### ISSUE TITLE
Having comment at end of tf file prevents outputs from being displayed

### ISSUE THREAD

**Issue by rberlind** [hashicorp/terraform#19224]
Having a comment at the end of may main.tf prevents outputs from being displayed after apply. But those outputs work fine and do show up if I don't have the comment.

### Terraform Version
```
terraform -v
Terraform v0.12.0-alpha1
```

### Terraform Configuration Files
I've only included the outputs from my main.tf, but complete code can be seen at https://github.com/rberlind/terraform-0.12-examples/tree/master/rich-value-types with difference that comment at bottom has been moved up which then does allow all outputs to be displayed.
```hcl
output "vpc" {
  value = module.network.vpc
}

output "subnet" {
  value = module.network.subnet
}

output "interface_sec_groups" {
  value = aws_network_interface.rvt.security_groups
}

/*output "instance_private_dns" {
  value = aws_instance.rvt.private_dns
}*/

```

### Crash Output
https://gist.github.com/rberlind/4dde903df93b1e0d8881c4756b5e6b00

### Expected Behavior
Several outputs should have been printed by apply.

### Actual Behavior
No outputs were printed.  However, if I remove the comment at the end of the file or put it above the outputs, then the outputs are printed.

However, I should also note that I tried a much simpler example and did not see this behavior: I used single main.tf with
```
output "first" {
  value = "first"
}

/*output "second" {
  value = "second"
}*/
```
In this case, running `terraform apply` did show me:

```
Outputs:

first = first
```

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

### Additional Context
complete code can be seen at https://github.com/rberlind/terraform-0.12-examples/tree/master/rich-value-types with difference that comment at bottom has been moved up which then does allow all outputs to be displayed

**Comment by apparentlymart** [comment]
By coincidence I was testing another configuration with multi-line (`/* ... */`) comments in it some time after we discussed this issue today and I saw a similar result that parts of the configuration seemed to be ignored when they were present. Once I removed those comments, the configuration was parsed correctly.

I suspect there's a parser bug in HCL causing this, so probably this will end up being an HCL PR as well as a Terraform PR; we'll take a closer look before v0.12.0 final.

**Comment by rberlind** [comment]
Thanks for the update @apparentlymart.

**Comment by apparentlymart** [comment]
This was fixed as a side-effect of #19537 because the underlying HCL issue was fixed and that update was included when we upgraded HCL for a different reason.

The problem here was that when a file had more than one multi-line comment in it HCL was treating the
whole span between the first `/*` and the final `*/` as a single comment, including any non-comment
content in between. Thus this problem was only visible in a file with at least two multi-line comments.
